pkgs/tools/X11/sunpaper/default.nix

```diff
@@ -0,0 +1,59 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, sunwait
+, wallutils
+, rPackages
+, pkgs
+}:
+
+stdenv.mkDerivation rec {
+  pname = "sunpaper";
+  version = "1.0";
+
+  src = fetchFromGitHub {
+    name = "sunpaper";
+    owner = "hexive";
+    repo = "sunpaper";
+    rev = "8d518dfddb5e80215ef3b884ff009df1d4bb74c2";
+    sha256 = "sCG7igD2ZwfHoRpR3Kw7dAded4hG2RbMLR/9nH+nZh8=";
+  };
+
+  buildInputs = with pkgs; [
+    wallutils
+    sunwait
+  ];
+
+  postPatch = ''
+    substituteInPlace sunpaper.sh \
+      --replace '$HOME/sunpaper/images/' "$out/images/"
+    substituteInPlace sunpaper.sh \
+      --replace '$HOME/.cache' "$TMPDIR/.sunpapercache"
+    substituteInPlace sunpaper.sh \
+      --replace "sunwait" "${sunwait}/bin/sunwait"
+    substituteInPlace sunpaper.sh \
+      --replace "setwallpaper" "${wallutils}/bin/setwallpaper"
```

Comment:
Maybe `wrapProgram $out/bin/sunpaper [...]`?

Reply:
We could at least only run `substituteInPlace` only once since it accepts the `--replace` flag several times :wink:

Reply:
From what I understand from the manual, the wrapper is useful if we are calling the program with different arguments. In this case, it's only replacing text a single time so what does `wrapProgram` add?

Reply:
> so what does `wrapProgram` add?

Well it's less reliant on the internal usage on the source and more of the actual behavior it's exhibiting, so even if sunpaper got updated and now they, I dunno, use some other program from `sunwait` or `wallutils` it'll still work without needing changes to this lil patch.

Reply:
However, `substituteInPlace` will warn if a replacement string isn't found, so by reading the build output you'll notice such change. With wrapProgram it won't warn because it just "wraps" the script and you'll may notice it breaking when running it instead.

Reply:
Well, this will be fine either way then so I guess we can leave it as is.

(it'd be nice if `substituteInPlace` had an option to fail if it can't find a string..)

Reply:
@etu @ckiee *phew* I was stressing trying to figure out how to make the wrapper work